### ISSUE TITLE
Astro magnitudes with flux density

### ISSUE THREAD

**Issue by mileslucas** [JuliaAstro/UnitfulAstro.jl#12]
I am having troubles applying magnitudes to units of flux density.

**Works**
```julia
using Unitful, UnitfulAstro

flux = @. randn(1000) * 1e-4 + 1e-3 
flux *= u"Jy"
mag = 3u"mag"
flux .* mag # equivalent to flux * 10^(-0.4 * mag)
```

**Doesn't work**
```julia
using Unitful, UnitfulAstro

flux = @. randn(1000) * 1e-4 + 1e-3 
flux *= u"erg/cm^2/s/cm" # Flux Density (eg. spectra)
mag = 3u"mag"
flux .* mag # Errors with error "error"
```

As far as I know, this should work, but the dimensions don't work out with Unitful.

**Comment by giordano** [comment]
Probably better to report to https://github.com/PainterQubits/Unitful.jl?

**Comment by mileslucas** [comment]
I'll try over there as well.

**Comment by mileslucas** [comment]
Closed with #13
